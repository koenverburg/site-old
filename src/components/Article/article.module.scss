@import '@styling/base';

.section {
  margin-top: 4rem;
  margin-bottom: 4rem;
}

.kicker {
  color: palette(blog-kicker);
  font-family: fontFamily(poppins);
  font-weight: fontWeight(medium);
  margin-top: 0;
  margin-bottom: 0.5rem;
}

.title {
  color: palette(blog-title);
  font-family: fontFamily(oswald);
  letter-spacing: 0.1rem;
  margin-top: 0;
  margin-bottom: 1.5rem;
}

.description {
  color: palette(blog-description);
  font-family: fontFamily(poppins);
  margin: 0;
}

.content {
  color: palette(blog-description);
  font-family: fontFamily(poppins);

  h1, h2, h3, h4, h5, h6, strong {
    color: palette(blog-title);
  }

  p {
    line-height: 1.9em;

    code {
      border-radius: 0.5rem;
    }
  }

  // needed for code blocks
  p > code,
  li > code {
    background: rgb(1, 22, 39);
    color: rgb(214, 222, 235);
    padding: 0.4em 0.3rem;
  }


  blockquote {
    margin: 0;
    border-left: 0.5rem solid palette(shark);

    p {
      margin: 0;
      text-indent: 1rem;
      background-color: rgba(palette(shark), 20%);
    }
  }

  hr {
    height: 0;
  }

  pre {
    padding: 0.5rem;
    border-radius: 0.5rem;
  }
}
