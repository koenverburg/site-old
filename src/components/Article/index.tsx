import Link from 'next/link'
import styles from './article.module.scss'

interface IProps {
  article: Object
}

export const Article: React.FunctionComponent = ({article, content}) => {
  return (
    <section className={styles.section}>
      <span className={styles.kicker}>{article.kicker}</span>
      <h1 className={styles.title}>{article.title}</h1>
      <article className={styles.content}>
        {content}
      </article>
    </section>
  )
}
