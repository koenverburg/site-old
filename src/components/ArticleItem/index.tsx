import Link from 'next/link'
import styles from './articleitem.module.scss'

interface IProps {
  article: Object
}

export const ArticleItem: React.FunctionComponent = ({article}) => {
  return (
    <Link href={`/blog/${article.slug}`} passHref>
      <a className={styles.root}>
        <span className={styles.kicker}>{article.kicker}</span>
        <h2 className={styles.title}>{article.title}</h2>
        <p className={styles.description}>{article.description}</p>
      </a>
    </Link>
  )
}
