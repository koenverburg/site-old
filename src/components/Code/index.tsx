import Highlight, { defaultProps } from 'prism-react-renderer'
import theme from 'prism-react-renderer/themes/nightOwl'
import styles from './code.module.scss'

export const Code = ({ codeString, language, metastring, ...props }) => {
  return (
    <Highlight
      {...defaultProps}
      code={codeString}
      language={language}
      theme={theme}
      {...props}
    >
      {({ className, style, tokens, getTokenProps }) => (
        <div className={styles.reactHighlight} data-language={language}>
          <pre className={className} style={style}>
            {tokens.map((line, i) => {
              return (
                <div>
                  {line.map((token, key) => (
                    <span {...getTokenProps({ token, key })} className={styles.token} />
                  ))}
                </div>
              )
            })}
          </pre>
        </div>
      )}
    </Highlight>
  )
}
