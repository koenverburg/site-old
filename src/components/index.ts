/* GENERATOR-IMPORT */
export {Icon} from './Icon'
export {ErrorBoundary} from './Errorboundary'
export {Layout} from './Layout'
export {ArticleItem} from './ArticleItem'
export {Article} from './Article'
export {Code} from './Code'
export {Menu} from './Menu'
