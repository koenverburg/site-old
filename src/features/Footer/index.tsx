import React from 'react'
import {gql} from '@apollo/client'
import {Social} from '@graphcms/types'
import styles from './footer.module.scss'
import {Icon} from '@components'

type Props = {
  socials: Social[],
}

const Footer: React.FunctionComponent<Props> = ({socials}) => {
  return (
    <div className={styles.footer}>
      <span className={styles.copyright}>Copyright {(new Date().getFullYear())}</span>
      {socials &&
        <ul className={styles.nav}>
          {socials.map(links => (
            <li key={links.id} className={styles.item}>
              <a
                role="button"
                aria-label={links.handle}
                href={links.url}
                target="_blank"
                rel="noopener"
              >
                <Icon name={links.icon} />
                {links.platform}
              </a>
            </li>
          ))}
        </ul>
      }
    </div>
  )
}

export const FooterFragments = {
  links: gql`
    fragment FooterLinks on Query {
      socials {
        id
        handle
        icon
        url
        platform
      }
    }
  `,
}

export {Footer}
