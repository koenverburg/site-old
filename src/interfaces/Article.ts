export interface IArticle {
  title: string
  subTitle?: string
  description: string
  kicker: string
  series?: string
  tags?: string[]
  slug: string
  date?: Date
  canonical_url?: string
}
