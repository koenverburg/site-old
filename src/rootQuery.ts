import {gql} from '@apollo/client'
import {FooterFragments} from '@fragments'

export const rootQuery = gql`
  query rootQuery  {
    ...FooterLinks
  }
  ${FooterFragments.links}
`
