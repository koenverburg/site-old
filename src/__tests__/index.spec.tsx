/* tslint:disable no-implicit-dependencies */
import React from 'react'
import App from '../pages/index'
import {render, RenderResult} from '@testing-library/react'
import {SignOff} from '../components/SignOff'
import {Article} from '../features/Article'
import {Footer} from '../features/Footer'
import {MockedProvider} from '@apollo/client/testing'
import {rootQuery} from '../rootQuery'

describe('App component', () => {
  let wrapper: RenderResult

  beforeAll(() => {
    const mocks = [
      {
        request: {
          query: rootQuery  ,
          variables: {},
        },
        result: {
          data: {
            // dog: { id: '1', name: 'Buck', breed: 'bulldog' },
          },
        },
      },
    ]

    wrapper = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <App />
      </MockedProvider>,
    )
  })

  it.skip('render', () => {
    expect(wrapper.container).toMatchSnapshot()
  })

  it.skip('should render a luckynumber componennt', () => {
    // const luckynumber = wrapper.container.firstElementChild.quer

    // expect(luckynumber).toBeDefined()
  })

  it.skip('should render a main element', () => {
    const mainEl =  wrapper.getElement()

    expect(mainEl).toBeDefined()
  })

  it.skip('the content section should include a Article component', () => {
    const articleComp = wrapper.find(Article)

    expect(articleComp).toBeDefined()
  })

  it.skip('the content section should include a Footer component', () => {
    const footerComp = wrapper.find(Footer)

    expect(footerComp).toBeDefined()
  })

  it.skip('the content section should include a SignOff component', () => {
    const signOffComp = wrapper.find(SignOff)

    expect(signOffComp).toBeDefined()
  })
})
