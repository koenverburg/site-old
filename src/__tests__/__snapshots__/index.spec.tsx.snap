// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component render 1`] = `
<main>
  <Section
    type="hero"
  >
    <LuckyNumber
      number="03"
    />
  </Section>
  <Section
    type="content"
  >
    <Article
      description="I'm a software engineer with a passion for automation, CI/CD and simplicity."
      name="Koen Verburg"
    />
    <Footer
      description="Below you will find a few platforms on which I'm active"
      handles={
        Object {
          "github": Object {
            "fa": Object {
              "iconName": "github",
              "prefix": "fab",
            },
            "handle": "@koenverburg",
            "url": "https://github.com/koenverburg",
          },
          "instagram": Object {
            "fa": Object {
              "iconName": "instagram",
              "prefix": "fab",
            },
            "handle": "@koenverburg_",
            "url": "https://instagram.com/koen.devops",
          },
          "medium": Object {
            "fa": Object {
              "iconName": "medium",
              "prefix": "fab",
            },
            "handle": "@koenverburg",
            "url": "https://medium.com/@koenverburg",
          },
          "pinterest": Object {
            "fa": Object {
              "iconName": "pinterest",
              "prefix": "fab",
            },
            "handle": "@thekoenverburg",
            "url": "https://pinterest.com/thekoenverburg",
          },
          "twitter": Object {
            "fa": Object {
              "iconName": "twitter",
              "prefix": "fab",
            },
            "handle": "@koenverburg_",
            "url": "https://twitter.com/koenverburg_",
          },
        }
      }
      title="Let's connect and talk code!"
    />
    <SignOff
      humanLinkText="This Stack"
      madeWith="Made with ❤️ in Rotterdam"
      poweredBy=" - powered by "
    />
  </Section>
</main>
`;
